planning/behavior_path_planner/src/path_utilities.cpp

```diff
@@ -36,14 +36,14 @@ namespace behavior_path_planner::util
 std::vector<double> calcPathArcLengthArray(
   const PathWithLaneId & path, size_t start, size_t end, double offset)
 {
+  start = std::max(start + 1, size_t{1});
+  end = std::min(end, path.points.size());
   std::vector<double> out;
+  out.reserve(end - start + 1);
 
```

Comment:
Let me give you a detailed comment since this is refactoring PR.

I prefer
```
std::vector<double> calcPathArcLengthArray(
  const PathWithLaneId & path, const size_t start, const size_t end, double offset)
{
  const size_t limited_start = std::max(start + 1, size_t{1});
  const size_t limited_end = std::min(end, path.points.size());
```
because if start, end is not const variable, they can be modified.
(I don't think the name `limited_start` is not so good though).

Moreover,
not
```
  const size_t limited_start = std::max(start + 1, size_t{1});
```
but
```
  const size_t limited_start = std::max(start, size_t{0});
```
with 
```
for (size_t i = limited_start + 1; i < limited_end; ++i) {
```
would be easier to read for me.

Reply:
Yes, I overlooked this. This is indeed nicer. Thank you for the suggestion!
```
  const size_t limited_start = std::max(start, size_t{0});
```

Reply:
i'll change limited to `bounded_start` and `bounded_end`. It provide better nuances in my opinion.

Reply:
DONE [593f95c](https://github.com/autowarefoundation/autoware.universe/pull/1832/commits/593f95c6ae7e1d6e0e29b710bc0b75e19f295570)